packages/react-admin/src/mui/input/ReferenceArrayInput.js

```diff
@@ -163,21 +164,29 @@ export class ReferenceArrayInput extends Component {
         }
     };
 
-    fetchReferences = ({ input, reference } = this.props) => {
+    fetchReferences = (props = this.props) => {
+        const { crudGetMany, input, reference } = props;
         const ids = input.value;
         if (ids) {
             if (!Array.isArray(ids)) {
                 throw Error(
                     'The value of ReferenceArrayInput should be an array'
                 );
             }
-            this.props.crudGetMany(reference, ids);
+            crudGetMany(reference, ids);
         }
     };
 
-    fetchOptions = ({ reference, source, resource } = this.props) => {
+    fetchOptions = (props = this.props) => {
+        const {
+            crudGetMatching,
+            reference,
+            source,
+            resource,
+            referenceSource,
+        } = props;
```

Comment:
why not using destructuring directly in the method signature ?

Reply:
because it's less legible when there is a default value and many props IMO